libs/blockmesh-cli/src/login_mode.rs

```diff
@@ -55,18 +57,43 @@ pub async fn login_mode(
         depin_aggregator,
         device_type: DeviceType::Cli,
     };
-    if is_ws_feature_connection().await? {
-        connect_ws(url, email, api_token, session_metadata).await?;
-    } else {
-        poll(url, email, api_token, session_metadata).await;
+    let mut prev_is_ws_feature = false;
+    let mut stop_notifier = Arc::new(Notify::new());
+    loop {
+        let is_ws_feature = is_ws_feature_connection().await.unwrap_or_default();
+        if is_ws_feature != prev_is_ws_feature {
+            stop_notifier.notify_one();
+        }
+        prev_is_ws_feature = is_ws_feature;
+        stop_notifier = Arc::new(Notify::new());
```

Comment:
why create a new notifier each loop? cant reuse?

Reply:
Yeah, it can be reused.